server/cli/src/test_connection.rs

```diff
@@ -0,0 +1,499 @@
+extern crate machine_uid;
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use clap::Parser;
+use log::*;
+use repository::{get_storage_connection_manager, migrations::Version};
+use reqwest::{Client, Url};
+use serde::{Deserialize, Serialize};
+use server::configuration;
+use service::{
+    apis::login_v4::{LoginApiV4, LoginInputV4, LoginUserTypeV4},
+    app_data::{AppDataService, AppDataServiceTrait},
+    service_provider::ServiceProvider,
+    settings::is_develop,
+    sync::{
+        api::{SyncApiSettings, SyncApiV5},
+        api_v6::SyncApiV6,
+        settings::SyncSettings,
+    },
+};
+use tokio::sync::mpsc;
+
+#[derive(Parser, Debug)]
+struct Args {
```

Comment:
I was wondering about the gui, is this tester meant to be run by users or admins ? If Admins don't really need gui if users then don't need command line arguments (enter in gui)

Reply:
good point - I had initially thought of this as a gui tool for users.. thinking that admins would already have things set up correctly. Now that it's done, I like the command line version better, but I'm reluctant to dump the gui now that it exists. 

I've changed so that if you run with command line args, then you get the command line version (admin usage) and if you just start the binary then you get the gui, with an admin & username input.